lua/entities/pewpew_base_cannon/init.lua

```diff
@@ -205,12 +209,16 @@ end
 function ENT:NewSystem_FireBullet()
 	local Dir, Pos = pewpew:GetFireDirection( self.Direction, self, ent )
 	local num = self.Bullet.Spread or 0
-	if (num and num != 0) then
-		local randomang = Angle( math.Rand(-num,num), math.Rand(-num,num), math.Rand(-num,num) )
-		Dir:Rotate(randomang)
+	if self.ShotCount then
+		for i=1, self.ShotCount do
+			if (num and num != 0) then
+				local randomang = Angle( math.Rand(-num,num), math.Rand(-num,num), math.Rand(-num,num) )
+				Dir:Rotate(randomang)
+			end
+			pewpew:FireBullet( Pos, Dir, self.Owner, self.Bullet, self, self.Direction )
+		end
 	end
-	pewpew:FireBullet( Pos, Dir, self.Owner, self.Bullet, self, self.Direction )
-	
+  
```

Comment:
Let's discuss the efficiency of this chunk and decide if we should look into some optimization

Reply:
the alternative is doing something like x = cos(time+i) and y = sin(time+i^2) or something to make it kinda look random.

Reply:
Maybe we could also pre-generate some random angles? I know that sounds weird but it could help for high fire rate multi-shot weapons

Reply:
seems odd, but so like a table of randomly generated angles made at the spawn of a cannon? Of which itll randomly pick angles from it, tbh that kinda sounds as bad

Reply:
its probably fine as long as we don't go too crazy with the number of shots
when i tested in lua 5.3 math.cos has about the same (or more) performance cost as math.random so i dont think that would be much better
(my benchmarking was fairly rudimentary so its possible it wasnt 100% accurate)